services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_user_services.py

```diff
@@ -171,28 +174,32 @@ async def progress_create_containers(
     # The entrypoint container name was now computed
     # continue starting the proxy
 
-    async for attempt in AsyncRetrying(
-        stop=stop_after_delay(
-            dynamic_services_scheduler_settings.DYNAMIC_SIDECAR_WAIT_FOR_CONTAINERS_TO_START
-        ),
-        wait=wait_fixed(1),
-        retry_error_cls=EntrypointContainerNotFoundError,
-        before_sleep=before_sleep_log(_logger, logging.WARNING),
-    ):
-        with attempt:
-            if scheduler_data.dynamic_sidecar.service_removal_state.was_removed:
-                # the service was removed while waiting for the operation to finish
-                _logger.warning(
-                    "Stopping `get_entrypoint_container_name` operation. "
-                    "Will no try to start the service."
+    try:
+        async for attempt in AsyncRetrying(
+            stop=stop_after_delay(
+                dynamic_services_scheduler_settings.DYNAMIC_SIDECAR_WAIT_FOR_CONTAINERS_TO_START
+            ),
+            wait=wait_fixed(1),
+            before_sleep=before_sleep_log(_logger, logging.WARNING),
+        ):
+            with attempt:
+                if scheduler_data.dynamic_sidecar.service_removal_state.was_removed:
+                    # the service was removed while waiting for the operation to finish
+                    _logger.warning(
+                        "Stopping `get_entrypoint_container_name` operation. "
+                        "Will no try to start the service."
+                    )
+                    return
+
+                entrypoint_container = await sidecars_client.get_entrypoint_container_name(
+                    dynamic_sidecar_endpoint=dynamic_sidecar_endpoint,
+                    dynamic_sidecar_network_name=scheduler_data.dynamic_sidecar_network_name,
                 )
-                return
-
-            entrypoint_container = await sidecars_client.get_entrypoint_container_name(
-                dynamic_sidecar_endpoint=dynamic_sidecar_endpoint,
-                dynamic_sidecar_network_name=scheduler_data.dynamic_sidecar_network_name,
-            )
-            _logger.info("Fetched container entrypoint name %s", entrypoint_container)
+                _logger.info(
+                    "Fetched container entrypoint name %s", entrypoint_container
+                )
+    except RetryError as err:
+        raise EntrypointContainerNotFoundError from err
```

Comment:
This inheritance is so strange. Are you sure about this? Since it actually works.
Also AssertionError is not based on RetryError and I'm sure it works all over the place

Reply:
I figured it out.

please use 
```python
from tenacity.retry import retry_if_exception_type
...
retry=retry_if_exception_type(EntrypointContainerNotFoundError),
```

instead of 
```python
retry_error_cls=EntrypointContainerNotFoundError,
```

Somehow I used the wrong pattern here.

Reply:
I think that yes you were using the wrong pattern, as in order to use retry_error_cls, then the class to use **must** inheritate from RetryError which makes sense.
but what you are now asking is different because:
- since you were abusing the system, the tenacity retry would raise EntrypointContainerNotFoundError instead of RetryError (just try it in a python console if you don't believe me) for **any error**
- if I change with what you say now, that means that if any other error raises it will go through and change the behavior, is that what you want?

Reply:
Correct, very good one here. Glad you spotted this.